### DISCUSSION TITLE
Suggestion for control network.

### DISCUSSION THREAD

**Discussion by soyatlas** [richrobber2/canvas-zoom#42] [Q&A] [answered]
With the advances in controlnet, it would be useful to be able to use canvas-zoom in the window on the left, where you make masks for the impaint.
The two modules together make for fabulous potential.

ControlNet could make the editing window much larger, and this is where Canvas-zoom comes into its own as it greatly improves the workflow.

Thank you all.

**Comment by daswer123** [comment]
@soyatlas 
Drawing in contorlNet is broken, it does not continue the result, but draws again.

I keep waiting for this one to be fixed, but so far no results. I don't use drawing in controlNet, so I don't know how much it's needed. I think I will try to implement, in any case, it will not be worse :)

You draw on controlNet, if canvas-zoom would work there, would that be useful?

https://user-images.githubusercontent.com/22278673/236293651-6bd20344-e2cd-4d56-bbff-bf8ccd6fb634.mp4

**Reply by soyatlas** [reply]
For me it is interesting, basically because I have a lack of eyesight and if I zoom in, it helps me a lot. It's just a suggestion. Besides, I'm at the beginning, the same thing I get better results in the impaint of the img2img tab.
The controlnet is under continuous review. Everything will be to notify them to see if they can review this work cycle.
In any case, everything looks fantastic.
Congratulations on the job. Whoops!
Thank you

**Comment by daswer123** [comment]


**Reply by soyatlas** [reply]
Fantastic work !
Thanks daswer123

**Comment by daswer123** [comment, marked as answer]
UPD
There was an update, now the integration with ControlNet must be enabled in the setting tab
Make sure you have more than 1 unit in controlNet

![image](https://github.com/richrobber2/canvas-zoom/assets/22278673/6f23d8b8-d0ec-447e-875e-66448a87477e)

**Reply by soyatlas** [reply]
Thanks !

Hi, I don't know if this is the right place to make this suggestion, and I don't know if it's possible either.

The idea is to have a draft to retouch what is marked. It is not always a solution to use the undo.

Thank you !

**Reply by daswer123** [reply]
If you mean something like an eraser, unfortunately not. From the beginning of the extension, I tried to think of something, but I do not have enough skills to implement it. Also that it would have to modify the gradio source code too much, which I would not want to do.

**Reply by soyatlas** [reply]
ah, ok. You try it. 
Thanks !

**Reply by Sasha91** [reply]
This option does not seem to work with the newest version of automatic1111

**Reply by daswer123** [reply]
Checked with 1.4 and 1.3.2 everything seems to work. Make sure you have more than 1 unit in ControlNet. If there is still a problem open issue

**Comment by daswer123** [comment]
Guys, there is a problem, if you have the built-in extension and cavnas-zoom enabled, then the zoom in controlNet will not work. So I recommend that you disable the built-in extension.

**Reply by daswer123** [reply]
Fixed, you can keep everything on and not worry about it
